### DISCUSSION TITLE
Allow opt-in advertising

### DISCUSSION THREAD

**Discussion by Biep** [element-hq/element-meta#2164] [Ideas]
### Your use case

My primary use case is poor Africans, who often cannot use social media because they don't have the money for the data connection. This way they could sell some personal data in an ethical way in order to gain more credits than they spend.
I often lose connection with my West-African contacts because they simply cannot afford to top up their credit, to mutual frustration, and this would remove that problem. (It would also be huge selling point for Matrix - being able to use it without going bankrupt! It might cause Matrix to spread like an oil slick.)

If it could be done by an integration, that would obviously be best - but I think some API support will be necessary.

#### What would you like to do?

Have an opt-in for advertisements, based in a user profile that the user controls.  A percentage of the money earned this way goes to the user; another percentage goes to Element development.
Text from unencrypted rooms can be used for targeting; encrypted rooms are (of course) off-limits.

#### Why would you like to do it?

This
- brings in money for Element development
- brings in money for Element users
- gives a venue for genuine advertisers, who may currently post off-topic messages
- brings Matrix under the attention of the commercial world
- allows poor users to use some commercial features of Element (e.g. support) by spending the earning thus gained.  There could even be some micro-support level that would be affordable for people with moderate Element use.
- creates a clear criterion for whether someone can be DM'ed for some commercial offer.  If the profile doesn't say "yes", such DM'ing is a clear violation of the rules, and the DMer can be banned for that reason.

#### How would you like to achieve it?

A user who opts in can manage a profile, sharing as much or as little as wanted (ideally with an indication of the difference in revenue following from certain choices, so as to make them more informed)
Advertisers can work with those profiles in an aggregate shape (see roughly how many people match a certain profile template, send publicity to all those matching a template, and so on).

**Maybe** this can be done by an integration - if the integration is able to read the non-encrypted rooms the user is in, and able to recognise the user among the room members.

Payment can be done by crypto - maybe using the system Brave uses.

### Have you considered any alternatives?

Not really - other than the integration mentioned above.

### Additional context

If this were done in a sufficiently modular way, other media could join in and use it, diminishing Google's grip on the advertising market and our personal information.

**Comment by Biep** [comment]
(Did *someone* move this to a discussion, or was that done automatically?)

This would go well with multiple accounts - so as to keep *some* privacy on the fact that one is a member of certain rooms.

**Comment by MTRNord** [comment]
This seems a like a horrible idea.

Just look at apps like Kik or instagram or facebook. These apps are mainly ads and barely usable for their original usecase. I think ads have absolutely no place in matrix or Element apps. 

Additionally payment services built into the element app itself seems like a bad idea. Especially considering the store rules of apple and android which want a revenue share too. 

Also lets not forget the ethical issues of crypto please....

TLDR: No. Just dont.

**Comment by Biep** [comment]
That is why it is **opt-in**.  People who don't activate it won't see any difference.
If it could be done through an integration that would be best.
And what ethical issues are there with Basic Attention Tokens?  I hadn't heard of any, but then I am not following the subject.

**Comment by msglm** [comment]
>This brings in money for Element development

I believe that [New Vector Ltd's business model of selling managed servers and support contracts](https://element.io/pricing) is already a respectable way of obtaining money while not intruding on user experience, user freedom, or user privacy. While all businesses need money, adding advertisements and solicitations to a product is just going hurt the product long-term, even if the advertising is "ethical".
>brings in money for Element users

I believe this is out of scope for a communication application. The product itself shouldn't focus on making people money, that's the job of the individual to do. If you want to sell services on the matrix protocol, even using Element, (either by selling server hosting, managing clients with the tool, or soliciting work from others using the protocol), to my knowledge, no restrictions exist on the tools themselves to prevent this, in the same way there's no restriction on using Email for making money. 
It's best that matrix and element stay tools of communication, and people wishing to make money leverage the communication aspects of the tool to do that, rather than by forcing advertising into the mix.
>gives a venue for genuine advertisers, who may currently post off-topic messages

This is a subject that webmasters and system administrators should deal with, and isn't something that should be enabled at a first-party application level. If a webmaster wishes to permit advertising and solicitation on his or her own server, that's fine. Even adding a server-unique feature for advertising is something that could be enabled as an integration or app-service is permissible. However, the developer's of element should spend their time on more pressing matters, and leave this work to individual server administrators to pursue.
>brings Matrix under the attention of the commercial world

I don't think adding adverts to your chatting application is the kind of publicity a protocol like Matrix needs. As a normal user, all I care about is quality of life of the ecosystem improving. Would adding advert integration really benefit the quality of life of your average user? I think not, the team should focus on something more real, like Decentralized Identities, encrypted metadata, or speeding up the server software by rewriting it in Golang.
>allows poor users to use some commercial features of Element (e.g. support) by spending the earning thus gained. 

Users of lower wealth already are better off using tooling like Element and Matrix due to it being free as in beer, and free as in freedom. Anyone can host the tool, anyone can sell the tool, anyone can learn the tool, and anyone can use the tool. The only barrier to entry, really, is a computer with conventional web-hosting ability and human agency.

Still, there are more useful ways to help out the economically disadvantaged. A well-moderated matrix room dedicated to the solicitation of work and job postings would be a good help to these people. I think efforts are best directed to something more conventional for aiding people of this class find money and work.

>There could even be some micro-support level that would be affordable for people with moderate Element use.

I'm certain that there are many system administrators and community members who would love to take some cash to help out in setting up and managing services! Users interested in this type of support are spoiled for choice, and often can get help for free from kind people, aiding people as volunteers in various chat-rooms on the matrix protocol.

>creates a clear criterion for whether someone can be DM'ed for some commercial offer. If the profile doesn't say "yes", such DM'ing is a clear violation of the rules, and the DMer can be banned for that reason.

I don't believe this can be enforced at a protocol or application level, and would end up falling on the effort of the system administrator who is managing a matrix instance. [Matrix already has this covered with some robust moderation tools](https://matrix.org/docs/communities/moderation/). 

>A user who opts in can manage a profile... Advertisers can work with those profiles...

This entire section seems like a very obtuse way of allowing the user to just subscribe to content their interested in. If a user is interested in information related to video games, then they can join a room in matrix dedicated to the posting of content related to that, including advertisement content. There's no need to have a separate party of advertisers attempt to categorize people in a streamlined fashion.

Even then, if you're an advertiser wishing to promote a product, you already can just join rooms people are interested in and collect information on them, that they post publicly. If a user is in a room dedicated to game discussions, posting information about games you're advertising in there would suffice, or inviting people to rooms dedicated to games, to rooms that promote game-related advertisements is fair (so long as your webmaster and law permits it).

>Payment can be done by crypto - maybe using the system Brave uses.

Again, this is a communication platform. If you wish to pay people in cryptocurrency, and broker using the Matrix protocol, I believe that is already robust enough to permit commerce between peers. If a system administrator wants to make a cryptocurrency-focused Matrix instance, they're free too.

>diminishing Google's grip on the advertising market and our personal information.

I'm going to delve out of the realm of neutrality with this one:
If your goal is to combat Google, then you're making a big mistake in the mentality for doing it. The issue with Google is not Google, nor their executive team, it's their behavior of engaging in manipulative, modern advertising, data collection, and [enshittification](https://en.wikipedia.org/wiki/Enshittification). I don't want a machine to dictate what I see, rather that machine is owned by Google, or that machine is operated off of a decentralized blockchain with the best accountability system known to man. I want to dictate what I see, because my machines are my tools, and I own them. I should have total agency to every aspect of that, and I don't care who benefits from the use of my tools or my attention, except who I deem to want to benefit. Algorithms for predicting my interests be damned.

I subscribe to RSS feeds for content I enjoy, I'm technically being "advertised to" by opting into [element.io's blog](https://element.io/blog/). I'm technically also advertised to by sitting in the discussion rooms for content I like, and being exposed to discussion about and alerts around this content. Neither of these compensate me with a blockchain token for my time, I subscribed to what I want to see, and am happy with it. I don't need to be paid as compensation for suffering through advertising, mainly because I am an independent agent in this transaction, and I am not suffering. If I don't want the content, I simply stop requesting it.

There's nothing wrong with making money, hell, there's nothing inherently wrong with cryptocurrency, but there is something wrong with this very "techbro" solution of importing the modern, centralized, manipulative internet into the decentralized, agency-focused paradigm that, in my eyes, Matrix (and other federated and decentralized protocols) are fundamentally about. I don't want to use my eyes, my time, and my computing power to look at ads, and if any system administrator implemented such a feature, i'd hope that moving away from their server would be easy and painless! The goal of a decentralized internet is to prevent organizations like Google, and their advertising obsession, from existing, either by moving to a more respectable model of profiteering, where you're paying for real-world labor and computing resources and not some phony intellectual property license, or you're freeloading off a charitable friend or colleague, who doesn't mind hosting things online just for the fun of it. 

Tl;dr:
Ads aren't necessary for running Matrix. Good money can be made off selling real labor and services on the protocol, or around the protocol, without adding any features currently. Even if you want to advertise, the protocol is a communication protocol, and nothing is stopping you from advertising as is, so long as your webmaster and local laws permit it. Streamlining it would be a pointless waste of resources that offer nothing to the user experience, except computational waste, and annoyance, when features that already exist fit the use-case well, and just require some human intervention to get moving.

**Comment by Biep** [comment]
Once again, ***It is opt-in***.  It is an option for people (such as myself) who **want** it.
Denying *some* people something because *others* would not want it is never a valid reason in my opinion.
Lots of people don't want the non-encrypted Etherpads - and there is a good case to be made that something inherently unencrypted and unencryptable doesn't sit well with the spirit of Matrix.  Is that a reason not to offer Etherpads?  ***No***.

Could we please discuss the proposal, and not some imaginary scenario where Matrix becomes advertisement-based?

**Reply by MTRNord** [reply]
You are literally asking for a protocol level ad system here. That's literally your proposal you made...

**Reply by Biep** [reply]
Literally?
> Maybe this can be done by an integration - if the integration is able to read the non-encrypted rooms the user is in, and able to recognise the user among the room members.

It seems that handing out thumbs down is easier than reading.  But then you *did* add **TL;DR**, so..

All I ask for on the system level is **at most** an opt-in API for accessing info in non-encrypted rooms.

**Reply by beedell-roke** [reply]
> All I ask for on the system level is at most an opt-in API for accessing info in non-encrypted rooms.[^1]

@Biep, what exactly is the API you need? As you state, an implementation would be the way to achieve this, so it isn't immediately obvious what the FR here is - do you want or need assistance creating the integration?

I've no interest in the supposed ethics or lack thereof. This interests me merely because it's an unusual use case that provides a potential for more traffic to servers, both:

*	Negatively, due to the amount of bandwidth that media-based advertisements necessitate, and:
*	Positively, due to user count increase from areas previously unable to access it.

I'm unfamiliar with Africa's political inaccess to digital communication services, but to my knowledge, WhatsApp and SMS are widespread. These do not currently include advertisements, so I fail to understand why Matrix differs.

[^1]: https://github.com/element-hq/element-meta/discussions/2164#discussioncomment-7410481:~:text=All%20I%20ask%20for%20on%20the%20system%20level%20is%20at%20most%20an%20opt%2Din%20API%20for%20accessing%20info%20in%20non%2Dencrypted%20rooms

**Comment by Biep** [comment]
My primary use case is poor Africans, who often cannot use social media because they don't have the money for the data connection.  This way they could sell some personal data in an ethical way in order to gain more credits than they spend.
I often lose connection with my West-African contacts because they simply cannot afford to top up their credit, to mutual frustration, and this would remove that problem.  (It would also be *huge* selling point for Matrix - being able to use it without going bankrupt!  It might cause Matrix to spread like an oil slick.)

If it could be done by an integration, that would obviously be best - but I think *some* API support will be necessary.

**Reply by MTRNord** [reply]
You sound like you want to sell water bottles like nestle and call yourself the lord and saviour tbh.
 https://www.theguardian.com/global/2018/oct/04/ontario-six-nations-nestle-running-water

**Reply by Biep** [reply]
Oh, thanks.  You mean the same Nestle that still owns the cocoa trees on the lands of the people in Ivory coast, forbidding them to cut them and plant a more profitable crop?  Wow.

But maybe, yes, in an inverted way.  Taking the "water" away from Google, and giving it to the local people to whom it originally belongs in my opinion.

**Comment by Biep** [comment]
But I admit the more general notion of taking advertising control away from Alphabet and its ilk and putting it in the hands of the end users is appealing to me.  Advertisers will go where it is cheapest, and as most of the Matrix space is non-profit, user ought to be able to undercut the commercial rates.  I am willing to let some advertiser use my general location for less money than Google would be - if only to take the advertiser away from Google.  And Google knows my general location from spying on me, whereas in this case I myself should be the person providing that information (or not) - or not opting in at all, of course.

The situation is not very different from identity servers.  Many people are against, and for good reason, so it is something to op in to.

**Comment by Biep** [comment]
I am merely a relationship counsellor working there - and doing most of the work through Matrix, for obvious reasons.  And seeking a way to enable the people there to contact me and the other counsellors I train.
But the use will be much wider.  People in villages want to need the city prices of crops, so as not to be cheated by the merchants, so having a means of communication *as such* helps earning them more.  And of course being able to stay in contact with family is highly valued in a communal society.
